### ISSUE TITLE
Register LiftService with DI using SimpleInjector rather than built-in asp.net DI.

### ISSUE THREAD

**Issue by ayrock-dev** [gymt/gymt-lift-service-proto#5]

**Comment by ayrock-dev** [comment]
@blthiewes Are you an informed and wise on such a topic?

**Comment by blthiewes** [comment]
a [dbcontext isn't thread safe](https://stackoverflow.com/questions/6126616/is-dbcontext-thread-safe). I register db context's as a scoped singleton. I think simpleinjector uses the term singleinstance instead of singleton anymore.

what do you mean by `if two controllers match a route`?

**Comment by ayrock-dev** [comment]
I guess I mean if a controller receives two concurrent requests before it's done with either. Does a db lock happen on the .NET side or on the database I/O side? Trying to get my head around why you want one Lifestyle over another.

**Comment by ayrock-dev** [comment]
On top of that, how do I accomplish a single instance Lifestyle? I couldn't figure out how to register a DbContext with SimpleInjector. I mentioned I'm actually just registering it with Asp.Net's DI and then enabling SimpleInjector's cross-wiring. Their docs said that will use the SimpleInjector container's DefaultScopedLifestyle though...

**Comment by blthiewes** [comment]
What database are you using? I believe it depends on that in terms of locking. 

with simple injector, you just new up an instance and register it as a `SingleInstance`.  I've never used Asp.Net's DI. I'm assuming cross wiring means you're using both at the same time?

**Comment by blthiewes** [comment]
did you follow https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-2.2&tabs=visual-studio ?

**Comment by ayrock-dev** [comment]
Yes, I did. Although we still on Core 2.1 not Core 2.2. Should be trivial upgrade since we don't have any code yet.

_What database are you using?_ - In memory database still. No decisions made yet on databases.

And - Yes I think cross-wiring means using both at same time. I can't figure out how to register a DBContext with SimpleInjector, haha. So I registered it with the built-in and SimpleInjector finds it with cross-wiring enabled :P
